2_decision_tree_gridsearch/Boosting_algorithms.md

```diff
@@ -0,0 +1,237 @@
+    print("F1 Score: ", f1_score(observed, y_pred, average='weighted'))
+    print("Precision Score: ", precision_score(observed, predicted, average='weighted'))
+    print("Recall Score: ", recall_score(observed, predicted, average='weighted'))
+```
+
+```python
+train, test = train_test_split(titanic_dataset, test_size=0.2, random_state=123)
+train_balanced = balance_undersample(train)
+train_copy = remove_over_95th(train_balanced, columns=['Age', 'Fare'])
+
+X,Y = exctract_target(train_copy)
+x_test, y_test = exctract_target(test)
+```
+
+```python
+columns_with_na_category = ['Embarked', 'Title']
+columns_with_na_numeric = ['Age']
+category_columns = ['Embarked', 'Sex', 'Title']
+columns_to_drop = ['PassengerId', 'Name', 'Ticket', 'Cabin']
+
+x_title = ExctractTitle().fit_transform(X)
+
+warnings.simplefilter(action='ignore', category=FutureWarning)
+target_encoders = ProcessTargetEncoding.fit_encoder(x_title, Y, category_columns)
+```
+
+```python
+preprocessing_steps = [
+   ('add_feature_title', ExctractTitle()),
+    ('add_feature_family_size', FamilySize()),
+    ('process_na_category', FillNaMode(columns_with_na_category)),
+    ('process_na_numeric', FillNaMode(columns_with_na_numeric)),
+    ('process_categories',  ProcessTargetEncoding(target_encoders, category_columns)),
+    ('drop_columns', DropColumns(columns_to_drop)),
+         ]
+preprocessing_pipeline = Pipeline(preprocessing_steps)
+```
+
+```python
+X_preprocessed = preprocessing_pipeline.fit_transform(X)
+x_test_processed = preprocessing_pipeline.fit_transform(x_test)
+```
+
+# AdaBoost
+
+```python
+ada_clf = AdaBoostClassifier(n_estimators=100, random_state=42)
+ada_clf.fit(X_preprocessed,Y)
+```
+
+```python
+y_pred = ada_clf.predict(x_test_processed)
+print_scores(y_test, y_pred)
+```
+
+```python
+ada_clf.score(x_test_processed, y_test)
+```
+
+```python
+param_grid = dict(
+    n_estimators = [50, 80, 100, 120, 200, 250, 300, 350],
+    learning_rate = [0.2, 0.3, 0.5, 0.8, 1.0, 1.5],
+    algorithm = ['SAMME', 'SAMME.R']
+)
+
+ada_grid_search = GridSearchCV(ada_clf, param_grid, n_jobs=-1)
+```
+
+```python
+EXPERIMENT_NAME = 'mlflow-ada_boost'
+EXPERIMENT_ID = mlflow.set_experiment(EXPERIMENT_NAME).name
+mlflow.sklearn.autolog(log_models=False, silent=True, max_tuning_runs=500)
+with mlflow.start_run() as run:
+    ada_grid_search.fit(X_preprocessed, Y)
+```
+
+```python
+ada_best = ada_grid_search.best_estimator_
+```
+
+```python
+ada_best
+```
+
+```python
+y_pred_a = ada_best.predict(x_test_processed)
+print_scores(y_test, y_pred_a)
+```
+
+<!-- #region tags=[] -->
+# Gradient boost
+<!-- #endregion -->
+
+```python
+grad_clf = GradientBoostingClassifier(n_estimators=100, learning_rate=1.0,
+     max_depth=1, random_state=42)
+grad_clf.fit(X_preprocessed,Y)
+```
+
+```python
+grad_clf.score(x_test_processed, y_test)
+```
+
+```python
+y_pred = grad_clf.predict(x_test_processed)
+print_scores(y_test, y_pred)
+```
+
+```python
+grad_param_grid = dict(
```

Comment:
the grid has weird param values. What you can try is having the grid of only the most influencial - `learning_rate, max_depth, n_estimators` and do in depth analysis of them (starting from values that would be obviously too low. check this for reference. https://medium.com/all-things-ai/in-depth-parameter-tuning-for-gradient-boosting-3363992e9bae .